src/math/exp_fcn.c

```diff
@@ -0,0 +1,216 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/*
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Shriram Shastry <malladi.sastry@linux.intel.com>
+ *
+ */
+
+/* Include Files */
+#include <sof/math/exp_fcn.h>
+#include <sof/math/numbers.h>
+#include <sof/common.h>
+#include <rtos/bit.h>
+#include <stdbool.h>
+#include <stdint.h>
+
+#define BIT_MASK_Q62P2 0x4000000000000000
+#define CONVERG_ERROR 28823037607936LL	// error smaller than 1e-4,1/2 ^ -44.7122876200884
+#define BIT_MASK_LOW_Q27P5 0x8000000
+#define MIN_IN_VAL -1342177280
+#define MAX_OUT_VAL 1244979733
+#define QUOTIENT_SCALE BIT(30)
+#define MAX_IN_VAL 1342177280
+#define TERMS_Q23P9 8388608
+#define MIN_OUT_VAL 56601
+#define LSHIFT_BITS 8192
+
+/* inv multiplication lookup table */
+/* LUT = ceil(1/factorial(b_n) * 2 ^ 63) */
+const int64_t iv[19] = {
```

Comment:
Now that this array is being re-used across multiple files, this name `iv` is visible across the entire binary. So it must be longer now to avoid collisions and to make it possible to find in the source code. Right now `git grep iv` returns more than 4000 lines.

Maybe something like `math_exp_inv[]`? Anything with some kind of namespace prefix.


Sorry I'm realizing this only now, should have said earlier.

Reply:
done